File: mycollab-core/src/main/java/com/esofthead/mycollab/core/cache/CacheEvict.java
/**
 * This file is part of mycollab-core.
 *
 * mycollab-core is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * mycollab-core is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with mycollab-core.  If not, see <http://www.gnu.org/licenses/>.
 */
package com.esofthead.mycollab.core.cache;

import java.lang.annotation.ElementType;
import java.lang.annotation.Inherited;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

/**
 * <b>NOTE: </b> Implement of cache just be presented in premium or ondemand
 * delivery.<br/>
 * This annotation denotes a method has data clean of cache with the key compose
 * by its arguments.
 * 
 * @author MyCollab Ltd.
 * @since 1.0
 * 
 */
@Inherited
@Retention(RetentionPolicy.RUNTIME)
@Target(value = { ElementType.METHOD })
public @interface CacheEvict {
}

File: mycollab-core/src/main/java/com/mycollab/core/cache/CacheEvict.kt
package com.mycollab.core.cache

import java.lang.annotation.*

/**
 * **NOTE: ** Implement of cache just be presented in premium or ondemand
 * delivery.<br></br>
 * This annotation denotes a method has data clean of cache with the key compose
 * by its arguments.
 *
 * @author MyCollab Ltd.
 * @since 1.0
 */
@Inherited
@Retention(AnnotationRetention.RUNTIME)
@Target(AnnotationTarget.FUNCTION, AnnotationTarget.PROPERTY_GETTER, AnnotationTarget.PROPERTY_SETTER)
annotation class CacheEvict
